chapters/en/slides/module7/module7_03.md

```diff
@@ -0,0 +1,389 @@
+here we sort the genres alphabetically before passing them to the
+dropdown.
+
+Since we recreated the `select_genre` variable, we need to also redefine
+the opacity encoding and add this new variable to overwrite the old one.
+
+Now you can use the dropdown to select movies of different genres!
+
+Note that the location of the dropdown is fixed and there is no way
+change its location.
+
+---
+
+## Setting the default value in a widget
+
+``` python
+select_genre = alt.selection_single(
+    fields=['Major_Genre'], bind=dropdown, init={'Major_Genre': 'Comedy'})
+
+points.add_selection(select_genre).encode(
+    opacity=alt.condition(select_genre, alt.value(0.7), alt.value(0.1)))
+```
+
+<iframe src="/module7/charts/03/unnamed-chunk-4.html" width="100%" height="420px" style="border:none;">
+</iframe>
+
+Notes: If we don’t specify a default value in a widget, the first value
+in the options array show up in the widget, but it is not used for
+selection until we interact with the dropdown.
+
+If we wanted to set a specific default value and immediately use it to
+highlight the data, we could specify which column and value to use via
+the `init` parameter as we have done in this slide.
+
+---
+
+## Adding multiple widgets to the same charts
+
+``` python
+mpaa_rating = sorted(movies['MPAA_Rating'].unique())
+dropdown_mpaa = alt.binding_radio(name='MPAA Rating ', options=mpaa_rating)
+dropdown_genre = alt.binding_select(name='Genre ', options=genres)
+select_genre_and_mpaa = alt.selection_single(
+    fields=['Major_Genre', 'MPAA_Rating'],
+    bind={'Major_Genre': dropdown_genre, 'MPAA_Rating': dropdown_mpaa})
+
+points.add_selection(select_genre_and_mpaa).encode(
+    opacity=alt.condition(select_genre_and_mpaa, alt.value(0.7), alt.value(0.1)))
+```
+
+<iframe src="/module7/charts/03/unnamed-chunk-5.html" width="100%" height="420px" style="border:none;">
+</iframe>
+
+Notes: Sometimes we want to highlight our data based on more than a
+single widget. For example, we might want to show only comedies that are
+rated as suitable for children to watch.
+
+We have already seen how to perform such breakdowns via faceting, but we
+could also do it by combining multiple interactive widgets.
+
+Here we’re adding a set of radio buttons for the maturity rating of the
+movies and a dropdown button for the genre selection.
+
+We create the radio buttons and dropdown separately, and then add then
+bind them both to the `selection_single` object to their respective
+fields.
+
+To trigger a selection, we now need to both click a radio button **and**
+make a selection in the dropdown.
+
+---
+
+## Adding sliders allows us to highlight points based on quantitative value comparisons
+
+``` python
+slider = alt.binding_range(name='Tomatometer ')
+select_rating = alt.selection_single(
+    fields=['Rotten_Tomatoes_Rating'],
+    bind=slider)
+
+points.encode(
+    opacity=alt.condition(select_rating, alt.value(0.7), alt.value(0.1))
+).add_selection(select_rating)
```

Comment:
Should we go back to full code here to remind folks where we are, as we have now dropped the radio buttons and legend selections? And should we keep the legend selections here? As again, there are a lot of categories...